src/cluster/etcd/watchmanager/manager.go

```diff
@@ -102,6 +110,7 @@ func (w *manager) watchChanWithTimeout(key string, rev int64) (clientv3.WatchCha
 	case <-doneCh:
 		return watchChan, cancelWatchFn, nil
 	case <-time.After(timeout):
+		cancelWatchFn()
 		err := fmt.Errorf("etcd watch create timed out after %s for key: %s", timeout.String(), key)
 		return nil, cancelWatchFn, err
```

Comment:
should it return `nil` instead `cancelWatchFn` here since it's already called?

Reply:
calling `nil` would cause a panic, and context cancellations are always idempotent in go code.